api/src/deployment.rs

```diff
@@ -314,6 +314,12 @@ impl JobQueue {
             queue.push(deployment.clone());
             debug!("loading deployment: {:?}", deployment.meta);
         }
+        debug!("sleeping for postgres to start...");
+
+        // Postgres takes a little bit of time to start up so we'll postpone
+        // the job processor loop until then.
+
+        tokio::time::sleep(std::time::Duration::from_secs(20)).await;
```

Comment:
Manually delaying the start of the loop like this may not be enough as that time would vary among environments and state of the underlying datadir.

Can we maybe use something like a connect with timeout and a test query to establish the cluster is ready instead? And after a global timeout if it didn't succeed, panic out so we can bubble up the error into the container's state.